### DISCUSSION TITLE
Sharing original dumps on the site

### DISCUSSION THREAD

**Discussion by Brume-AL** [atari-legend/atari-legend#212] [Ideas]
Hi all,

Well, time has come to share all dumps I made since 10 years. 😉

I've dumped 3000 games, compilations, educational games, and some tools/profesionnal software. All are original disks from my own collection. Haven't counted them all, but I think there's something like 5000 or 6000 floppy disks. All have been archived in Kryoflux  formats (.RAW and .CTR files) and / or SuperCard Pro format (.SCP format). Of course they can be easily converted in .STX or .HFE formats, which is more common for users. But keep in mind CTR and SCP can be used with emulators like Steem or Hatari.

So I suggest to upload them in all formats on Atari Legend, not only in .STX format. It means the site will host a lot of files, but I'd like to share all my work if possible. Also I'll have to check each disk wth Aufit before uploading to avoid bad dumps, despite I already did that when I dumped each floppy. But well, two verifications is better than one. 😊
It will be a huge job, so don't expect to see the 5000 disks online very soon. But I could do it step by step. 

Before starting the upload, maybe we could discuss some improvment of the download page. For instance, I'd like very much to display a view of each disk (the view is generated with Aufit). It looks like this:
![Donald Duck's Playground](https://user-images.githubusercontent.com/12120244/218263131-598aa156-8997-4695-aa45-cbc116e79b16.png)
We could also display some information for each dump:
- the kind of hardware used (Kryoflux, SuperCard Pro, Greaseweazle, etc)
- the parameters used (number of revolutions, index or splice mode, etc.).
- the date it was made
- the name of the dumper (I hope I won't be alone).

Let me know what you think about it, if you have something else in mind... And of course, if this kind of features could be easily added to the site (@nguillaumin 😉).

Thanks for reading & have a great week-end!

Bruno

**Comment by stgraveyard** [comment]
This is beautiful work Bruno. We talked about this at STNICCC back in 2015. I'm a big fan of doing everything very structured so I love this stuff. I wonder how this all fits in with what is already uploaded, and if we need to start all over from scratch or not (maybe that is me being authistic again). I'm looking forward to this project and am curious as to what others have to say.

**Reply by nguillaumin** [reply]
I don't think we have to start from scratch. The DB has been designed so that multiple dumps can be associated for a given disk, so we can keep what we have and add to it. Then later if we want to we can clean up the older dumps.

**Comment by nguillaumin** [comment]
Great! It would be awesome to have more dumps on the site, of course!

The thing is that currently uploading dumps is cumbersome in the legacy CPANEL. It takes many clicks to create the "media", then upload the dumps. It's a part I want to rewrite in the new CPANEL to be more ergonomic and faster. I'm thinking drag'n drop of multiple files at once and an easy way to associate them with the games/release. In your case you have so many that it might be even beneficial that you upload them in a folder on the server, and we have the CPANEL scan that folder and allow a quick import by specifying the game/release to associate the files with.

Aufit: Yes it would be great to have these and we can update the site to show them. I looked if we could generate them automatically but unfortunately it looks like the [open-source version of Aufit](https://github.com/DrCoolzic/Aufit) does not include the code to generate the images. I don't know if you have any connection with the author, but if they could share their source then we could have CPANEL generate the Aufit image automatically when the dumps are uploaded. If not then we need an additional upload endpoint in the CPANEL to submit the Aufit images.

Adding more dump information: Yes, no problem. However I would do it only on the new CPANEL, meaning I need to rewrite this part first. Currently the new CPANEL does not allow editing releases, so even less dumps, so there's a bit of work for me here, but I can start looking into it.

Could you share an example list of files you have? It would help me understand the structure and all files associated with a game, so that I can implement it suitably in CPANEL. I'm not very familiar with all the dump formats.

**Reply by nguillaumin** [reply]
I made some progress in CPANEL and I'm in the process of re-implementing the dump upload screens, with ergonomic (hopefully) drag'n drop.

Also good news, I think I found a way to automatically generate the disk track views, using the open source HxC floppy emuator tools. An example:

![test](https://github.com/atari-legend/atari-legend/assets/762512/37b65037-de1b-4e0c-ad02-ff8f027d9063)

So it should be possible to generate these automatically when a flux-based image is uploaded in CPANEL.

I'm still interested @Brume-AL by an example of list of files that you would typically upload for a set of games, so that I can see if I could build a way to bulk-upload multiple dumps/games in one go.

**Reply by nguillaumin** [reply]
This is now deployed on the DEV site if you want to have a look. The CPANEL supports uploading dumps and the track pictures will be displayed on the main site, for example: https://dev.atarilegend.com/games/release/564

The pictures are not generated when the dump is uploaded as it's too resource intensive, instead they are generated every hour with a scheduled task, for newly updated dumps.

**Reply by stgraveyard** [reply]
This looks amazing Nico. It is not a subject I am knowledgeable about, but I haven't seen this on any Atari site. I will let the team know. Thanks so much.
